app/graphql/mutations/create_asset.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class Mutations::CreateAsset < Mutations::BaseMutation
+  argument :signed_blob_id, String, "Signed blob ID generated via `createDirectUpload` mutation", required: true
+  argument :parent_id, ID, required: true
+
+  field :asset, Types::DigitalObject::AssetType, null: true
+
+  def resolve(signed_blob_id:, parent_id:)
+    parent = DigitalObject::Item.find(parent_id)
+    ability.authorize! :create_objects, parent.primary_project
+
+    blob = ActiveStorage::Blob.find_signed(signed_blob_id)
+
+    asset = initialize_child_asset(parent, blob.filename.to_s)
+
+    begin
+      asset.resources['master'] = initialize_asset_resource_for_blob(asset, blob, Hyacinth::Config.resource_storage)
+      asset.save!
+      parent.append_child_uid(asset.uid)
+      parent.save
+      { asset: asset }
+    ensure
```

Comment:
Ignore this if I'm wrong, but it appears that if initialize_asset_resource_for_blob successfully copies the upload to our final preservation storage location, and then `asset.save!` fails, the newly copied file won't be cleaned up?

Note that deleting an asset will not destroy it's master resource file on disk, since we definitely wouldn't want to do that for files that "reference" their preservation storage location on import (e.g. files in a bag).

Reply:
That's true - but I'm also not sure how to fix it. Change `Asset.destroy` to clean up resources if the resource storage service has the location uri?

Reply:
Add a third thing to the `ensure` block?

Reply:
Yeah, since this is JUST for asset creation via file upload, adding a third thing to the `ensure` block sounds good.